src/domain/tokens/useInfoTokens.ts

```diff
@@ -0,0 +1,219 @@
+import { getTokens, getWhitelistedTokens } from "./utils";
+import { getContract } from "../../config/contracts";
+import useSWR from "swr";
+import { contractFetcher } from "../../lib/contracts";
+import VaultReader from "../../abis/VaultReader.json";
+import {
+  BASIS_POINTS_DIVISOR,
+  DEFAULT_MAX_USDG_AMOUNT,
+  MAX_PRICE_DEVIATION_BASIS_POINTS,
+  USD_DECIMALS,
+  USDG_ADDRESS,
+} from "../../lib/legacy";
+import { getServerUrl } from "../../config/backend";
+import { Token, TokenInfo } from "./types";
+import { BigNumber } from "ethers";
+import { Web3ReactContextInterface } from "@web3-react/core/dist/types";
+import { bigNumberify, expandDecimals } from "../../lib/numbers";
+
+export function useInfoTokens(
+  library: Web3ReactContextInterface["library"],
+  chainId: number,
+  active: boolean,
+  tokenBalances: BigNumber[] | undefined,
+  fundingRateInfo: BigNumber[] | undefined,
+  vaultPropsLength: number | undefined
+) {
```

Comment:
Added, and, I realized that if look in detail, in TokenInfo almost all fields are optional, because now the code is written so that you can get TokenInfo without waiting for the request and then there will only be fields from the config